### PR TITLE
Dash canvas bugs

### PR DESCRIPTION
**Pull request by ghsolomon**
Hoist P/R Checklist
-------------------

**Pull request authors:** Review and check off the below. Items that do not apply can also be
checked off to indicate they have been considered. If unclear if a step is relevant, please leave
unchecked and note in comments.

- [x] Caught up with `develop` branch as of last change.
- [x] Added CHANGELOG entry, or determined not required.
- [x] Reviewed for breaking changes, added `breaking-change` label + CHANGELOG if so.
- [x] Updated doc comments / prop-types, or determined not required.
- [x] Reviewed and tested on Mobile, or determined not required.
- [x] Created Toolbox branch / PR, or determined not required.

**If your change is still a WIP**, please use the "Create draft pull request" option in the split
button below to indicate it is not ready yet for a final review.

> **Pull request reviewers:** when merging this P/R, please consider using a *squash commit* to
> collapse multiple intermediate commits into a single commit representing the overall feature
> change. This helps keep the commit log clean and easy to scan across releases. PRs containing a
> single commit should be *rebased* when possible.

### CODE REVIEW

File: desktop/cmp/dash/canvas/DashCanvasModel.js [hunk 1]
```diff
@@ -168,6 +170,14 @@ export class DashCanvasModel extends HoistModel {
             track: () => [this.viewState, this.layout],
             run: () => this.publishState()
         });
+
+        this.addReaction({
+            when: () => this.ref.current,
+            run: () => {
+                const {current: node} = this.ref;
```
Comment (lbwexler) [inline desktop/cmp/dash/canvas/DashCanvasModel.js:177]: find the destructure with aliasing a little ponderous -- would like to discuss this vs.
`const node = this.ref.current;`

Comment (lbwexler) [reply desktop/cmp/dash/canvas/DashCanvasModel.js:177]: also, in this case, node can just be the argument to the run function.

File: desktop/cmp/dash/canvas/DashCanvasModel.js [hunk 2]
```diff
@@ -338,6 +348,14 @@ export class DashCanvasModel extends HoistModel {
         layout = layout.map(({i, x, y, w, h}) => ({i, x, y, w, h}));
         if (!isEqual(this.layout, layout)) {
             this.layout = layout;
+
+            // Check if scrollbar visibility has changed, and force resize event if so
+            const {current: node} = this.ref,
```
Comment (lbwexler) [inline desktop/cmp/dash/canvas/DashCanvasModel.js:353]: are we sure that node exists here?

File: desktop/cmp/dash/canvas/DashCanvasModel.js (hunk 1 above)
Comment (ghsolomon) [reply desktop/cmp/dash/canvas/DashCanvasModel.js:177]: Sounds good- I see what you're saying.  I think I tried using `node` as the argument to the `run` function and it not working, but will double-check.  I assumed that maybe it was only passed to `run` when using `track` (and not `when`)

**Comment by ghsolomon** [comment]
#2959

**Pull request merged**